src/commands/lint/mod.rs

```diff
@@ -0,0 +1,118 @@
+pub use args::*;
+#[cfg(feature = "rustup")]
+use serde::Deserialize;
+
+#[cfg(feature = "rustup")]
+use crate::external_cli::Package;
+
+mod args;
+
+/// Represents the contents of `rust-toolchain.toml`.
+#[cfg(feature = "rustup")]
+#[derive(Deserialize)]
+struct RustToolchain {
+    toolchain: Toolchain,
+}
+
+#[cfg(feature = "rustup")]
+#[derive(Deserialize)]
+struct Toolchain {
+    channel: String,
+}
+
+/// Runs `bevy_lint`, if it is installed, with the given arguments.
+///
+/// Calling `lint(vec!["--workspace"])` is equivalent to calling `bevy_lint --workspace` in the
+/// terminal. This will run [`find_bevy_lint()`] to locate `bevy_lint`.
+pub fn lint(args: &mut LintArgs) -> anyhow::Result<()> {
+    use anyhow::ensure;
+
+    #[cfg(feature = "web")]
+    use crate::commands::lint::args::LintSubcommands;
+    use crate::{bin_target::select_run_binary, config::CliConfig, external_cli::cargo};
+
+    let auto_install = args.auto_install();
+
+    let metadata = cargo::metadata::metadata()?;
+
+    let mut bin_target = select_run_binary(
+        &metadata,
+        args.cargo_args.package_args.package.as_deref(),
+        args.cargo_args.target_args.bin.as_deref(),
+        args.cargo_args.target_args.example.as_deref(),
+        args.target().as_deref(),
+        args.profile(),
+    )?;
+
+    let config = CliConfig::for_package(
+        &metadata,
+        bin_target.package,
+        args.is_web(),
+        args.is_release(),
+    )?;
+
+    args.apply_config(&config);
+    // Update the artifact directory based on the config, e.g. in case the `target` changed
+    bin_target.update_artifact_directory(
+        &metadata.target_directory,
+        args.target().as_deref(),
+        args.profile(),
+        args.cargo_args.target_args.example.is_some(),
+    );
+
+    #[allow(
+        unused_mut,
+        reason = "This only needs to be mutable when the web feature is enabled"
+    )]
+    let mut cargo_args = args.cargo_args_builder();
+
+    #[cfg(feature = "web")]
+    if matches!(args.subcommand, Some(LintSubcommands::Web)) {
+        use crate::web::profiles::configure_default_web_profiles;
+
+        let mut profile_args = configure_default_web_profiles(&metadata)?;
+        // `--config` args are resolved from left to right,
+        // so the default configuration needs to come before the user args
+        profile_args.append(&mut args.cargo_args.common_args.config);
+        args.cargo_args.common_args.config = profile_args;
+
+        cargo_args = args.cargo_args_builder();
```

Comment:
(Nit) When the `web` feature is enabled, `args.cargo_args_builder()` is run, never read, then ovewritten by another call to `args.cargo_args_builder()`. It's basically calling the same function twice but only using the second result. Can this be simplified a bit more to avoid the duplication?